x/zoneconcierge/keeper/ibc_packet_btc_headers.go

```diff
@@ -0,0 +1,39 @@
+package keeper
+
+import (
+	"context"
+
+	"github.com/babylonlabs-io/babylon/x/zoneconcierge/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// BroadcastBTCHeaders sends an IBC packet of BTC headers to all open IBC channels to ZoneConcierge
+func (k Keeper) BroadcastBTCHeaders(ctx context.Context) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+	// 1. Check channels first - fail fast
+	openZCChannels := k.GetAllOpenZCChannels(ctx)
+	if len(openZCChannels) == 0 {
+		k.Logger(sdkCtx).Info("no open IBC channel with ZoneConcierge, skip broadcasting BTC headers")
+		return
+	}
+
+	// 2. Get headers to broadcast
+	headers := k.getBTCHeadersToSend(ctx, types.FullChainFetch)
```

Comment:
This will be likely to bloat the IBC packet tbh, as the IBC packet will contain the entire BTC header chain.

Instead, it shall start from the base header of the BTC light client in the contract side. For now wdyt we still use `WDeepFetch` mode?

After we have proper initialisation of BTC light client in contract, we can revisit this part

Reply:
> Shouldn't the base header in Babylon and consumer contract match?

Not really, because Babylon and BSN might not launch at the same time. If the BSN launches 1 years after Babylon then it can use a base header much later than Babylon

> Can we send multiple smaller IBC packets (in the same EndBlocker) instead of a big one?

Ideally, the BSN will use a base header that is slightly earlier to its launch date (e.g., 100 headers prior to the launch). Otherwise it adds overhead to both Babylon IBC traffic and the storage overhead of BSN

Reply:
Yes sounds good on reusing the old [headersToBroadcast](https://github.com/babylonlabs-io/babylon/blob/07836b55989afef5400548d410cb93c0fb813792/x/zoneconcierge/keeper/ibc_packet_btc_timestamp.go#L165) for this PR, and fix the contract init as a subsequent task